src/components/Card.js

```diff
@@ -35,27 +56,59 @@ export default class Card {
     this._cardCaption.textContent = this._title;
   };
 
-  generateCard() {
+  _setLikeStatus = () => {
+    if (this._isLiked) {
+      this.likeCard();
+    } else {
+      this.dislikeCard();
+    }
+  };
+
+  _toggleLike = () => {
+    this._cardLikeButton.classList.toggle("card__heart-button_liked");
+    this._isLiked = !this._isLiked;
```

Comment:
You need to **toggle likes** only in block `then` after a successful response from the server because if there is an error the user will not know about that and will see the toggled heart color.